core/dbt/cli/example.py

```diff
@@ -0,0 +1,25 @@
+import click
+import sys
+
+from dbt.cli.main import cli as dbt, deps
+
+
+def make_context(args):
+    try: 
+        ctx = dbt.make_context('dbt', args)
+    except click.exceptions.Exit:
+        exit()
+
+    ctx.invoked_subcommand = ctx.protected_args[0] if ctx.protected_args else None
+    return ctx
+
+# python core/dbt/cli/example.py
+# python core/dbt/cli/example.py --version
+# python core/dbt/cli/example.py deps --project-dir <project-dir-path>
+if __name__ == "__main__":
+    ctx = make_context(sys.argv[1:])
```

Comment:
@ChenyuLInx - Idea here being that `make_context` could be passed in objects to add to `ctx.obj`, bypassing/modifying what's configured by the CLI.

Reply:
Should the example be constructing args explicitly rather than trying to take sys.args? given that we are trying to show other teams how to use the interface programmatically?

Reply:
I think constructing context objects like a Project or Profile (or an EventHandler with registered callbacks once we have that!) would be useful, yes! `sys.args` is just for the CLI arguments.